### PR TITLE
[REVIEW] Enable more `fixed_point` unit tests by introducing "scale-less" constructor [skip ci]

### PR DESCRIPTION
**Pull request by codereport**
Address one of the bullet points in issue https://github.com/rapidsai/cudf/issues/3556

In https://github.com/rapidsai/cudf/pull/5704, the type list `FixedWidthTypesWithoutFixedPoint` was added for the unit tests that did not work out of the box when the type list `FixedPointTypes` was added to the type list `FixedWidthTypes`. About 40% of the unit tests that used `FixedWidthTypes` worked - the other 60% tend not to work because they construct `fixed_width_column_wrappers` from integral or floating point types, but `fixed_point` types cannot currently be constructed from just a single value - you need to specify both the value and the scale explicitly. By introducing a "scale-less" construct, we will be able to enable many more of the existing unit tests. The constructor will look something like below:

```cpp
  template <typename T,
            typename simt::std::enable_if_t<is_supported_construction_value_type<T>()>* = nullptr>
  CUDA_HOST_DEVICE_CALLABLE fixed_point(T const& value)
    : _value{static_cast<Rep>(value)}, _scale{scale_type{0}} {}
```
Also, decided for the tests (`factory_tests` and `fill_tests`) that require changes to `make_*_column`, to leave this to the follow up PR that moves `scale` to `data_type`)

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/include/cudf/fixed_point/fixed_point.hpp [hunk 1]
```diff
@@ -311,6 +311,17 @@ class fixed_point {
   CUDA_HOST_DEVICE_CALLABLE
   explicit fixed_point(scaled_integer<Rep> s) : _value{s.value}, _scale{s.scale} {}
 
+  /**
+   * @brief "Scale-less" constructor that constructs `fixed_point` number with a speficied
```
Comment (harrism) [inline cpp/include/cudf/fixed_point/fixed_point.hpp:315]:
```suggestion
   * @brief "Scale-less" constructor that constructs `fixed_point` number with a specified
```

Comment (codereport) [reply cpp/include/cudf/fixed_point/fixed_point.hpp]: Done :)

**Comment by codereport** [comment]
> Approve, just one typo.
> 
> Can you estimate what fraction of tests still need updates for fixed-point?

`FixedWidthTypes` + `FixedWidthTypesWithoutFixedPoint` are used 43 times in total, before this PR the split was:

||`FixedWidthTypes`|`...WithoutFixedPoint`|`TOTAL`|`% Left`|
|:-:|:-:|:-:|:-:|:-:|
|Before this PR|17|26|43|60%|
|After this PR|28|15|43|35%|

Two of them I have the fixes for but it will be better to fix them after in the `Move scale to data_type` PR as they involve modifications to the `make_*_column` factory functions. Two of the others are `clamp` and `copy_if_else` which can be done in separate followup PRs. The rest are mostly aggregation tests yet to be supported and one of the instances is the declaration which we can remove once all are supported.

**Pull request merged**